src/video.tsx

```diff
@@ -55,28 +102,61 @@ const Video = React.forwardRef<HTMLVideoElement, VideoProps>(
             autoPlay = !isRunningAcceptanceTest(), // default true, but disable autoPlay in screenshot tests
             muted = true,
             loop = true,
-            preload = 'none',
+            preload = 'auto',
+            loadingTimeout = 10000,
+            onError,
+            onPause,
+            onPlay,
             aspectRatio = '1:1',
             dataAttributes,
             ...props
         },
         ref
     ) => {
         const borderRadiusContext = useMediaBorderRadius();
+        const [isLoadComplete, setIsLoadComplete] = React.useState(false);
+
+        const [videoStatus, dispatch] = React.useReducer(videoReducer, 'loading');
+        const [currentVideoStatus, setCurrentVideoStatus] = React.useState<VideoState>('loading');
```

Comment:
why is this state repeated?

Reply:
discussed offline and changed logic to avoid unnecessary states.